### PR TITLE
VideoPress: improve blocks building process

### LINKED ISSUE #28016: VideoPress: improve building system
A recent change exposed a problem with the building of the VideoPress plugin. The final bundle is relying on 2 files from the `src/client` directory:

- `$videopress_extensions_file = __DIR__ . '/client/block-editor/extensions/index.json';` on [class-block-editor-extensions.php](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/class-block-editor-extensions.php)
- `$videopress_video_metadata_file = __DIR__ . '/client/block-editor/blocks/video/block.json';` on [class-initializer.php](https://github.com/Automattic/jetpack/blob/01e736f4025a81f25e212e1b97992a1d9dff9d33/projects/packages/videopress/src/class-initializer.php)

This caused the block to not appear on the block editor after we removed `src/client` from the final bundle (#27926).

We [brought back the client source to the bundle](https://github.com/Automattic/jetpack/pull/27979/commits/0480a15211875b033b27c0930300bef2c1566de5) to tentatively solve the issue, but now it's time to fix the building process to:

- make sure we have the mentioned JSON files on the `build` folder
- reference only the `build` code on the final bundle
- not bundle the `src/client` folder

### PR DESCRIPTION
**Pull request by retrofox**
Fixes https://github.com/Automattic/jetpack/issues/28016

#### Changes proposed in this Pull Request:

* Add `block.json` files to the `build`directory, for every block registered on the plugin
* Add `extensions/index.json` file to the `build` directory
* Update code references to the JSON files so they point to the `build` version
* Remove `src/client` code from the final plugin bundle
* Integrate extended block scripts

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

#### Testing instructions:

* Launch a new JN site with VideoPress standalone plugin and connect it (you can use the free plan)
* Go to the editor
* Look for the VideoPress block:

Screen Shot 2022-12-20 at 16 31 15

* Confirm that you can add the block to the editor
* Confirm that you can upload one video and use it
* Confirm that you can edit the video informations on the right side sidebar:

Screen Shot 2022-12-20 at 16 34 53

* In other words, confirm that block is present and working as expected
* Do the same testing on WPCOM

### CODE REVIEW

File: projects/packages/videopress/webpack.config.js [hunk 1]
```diff
@@ -30,6 +40,15 @@ module.exports = [
 			...jetpackWebpackConfig.StandardPlugins( {
 				DependencyExtractionPlugin: { injectPolyfill: true },
 			} ),
+			new CopyWebpackPlugin( {
+				patterns: [
```
Comment (lhkowalski) [inline projects/packages/videopress/webpack.config.js:44]: I think we may need a pattern to copy the `extensions/index.json` file as well. What do you think?

Comment (lhkowalski) [reply projects/packages/videopress/webpack.config.js:44]: I added it.

**Comment by lhkowalski** [comment]
I can confirm that now the block is being showed on a fresh JN site:

*With Jetpack:*

Screen Shot 2022-12-20 at 15 50 48

*Only VideoPress standalone:*

Screen Shot 2022-12-20 at 15 53 25

I will update the PR description to add testing instructions so we can get some feedback.

**Comment by lhkowalski** [comment]
Tested on WPCOM, the block is being showed now. The block features seems to work as well:

Screen Shot 2022-12-20 at 17 17 56

File: projects/packages/videopress/package.json [hunk 2]
```diff
@@ -80,6 +80,7 @@
 		"@wordpress/icons": "9.13.0",
 		"@wordpress/url": "3.23.0",
 		"classnames": "2.3.1",
+		"copy-webpack-plugin": "11.0.0",
```
Comment (anomiex) [inline projects/packages/videopress/package.json]: [nitpick] This should probably go in the devDependencies, along with other packages that are used for the build rather than the functionality. Not that it makes a huge difference.

**Review by anomiex — COMMENTED** [review]
Code changes look sensible and build output appears to be as intended. Haven't tested it myself.

One nitpick inline, feel free to ignore if you disagree.

**Comment by retrofox** [comment]
### JN with VideoPress standalone plugin

- [x] Connect the site
- [x] In the block editor, create a new VideoPres video block
- [x] Transform from VideoPress video block to Core embed
image
- [x] Transform from Core embed to VideoPress video

image

### JN with Jetpack plugin

- [x] Connect the site
- [x] Activate VideoPress module
- [x] In the block editor, confirm v6 is not available
- [x] Enable Beta blocks
- [x] In the block editor, create a new VideoPres video block
- [x] Convert the Embed block and vice versa (same as VideoPress video block)
- [x] Convert VideoPress video to Core Video and vice versa

image

### Simple site testing

- [x] Confirm v6 is not available when unproxied
image

- [x] Create a core/video instance
- [x] Confirm it converts to VideoPress video instance on the fly (when proxied)

image

- [x] Confirm it doesn't convert Core Embed videos on the fly
- [x] Convert VideoPress video to Core Video and vice versa

**Review by lhkowalski — APPROVED** [review]
@retrofox excellent job on the extensions code refactoring :)
